platform/security/sts/security-sts-pkivalidator/src/main/java/org/codice/ddf/security/validator/pki/PKITokenValidator.java

```diff
@@ -204,10 +208,40 @@ public TokenValidatorResponse validateToken(TokenValidatorParameters tokenParame
       Credential returnedCredential = validator.validate(credential, requestData);
       X500Principal subjectX500Principal =
           returnedCredential.getCertificates()[0].getSubjectX500Principal();
+
       response.setPrincipal(subjectX500Principal);
       if (response.getAdditionalProperties() == null) {
         response.setAdditionalProperties(new HashMap<>());
       }
+
+      try {
+        if (returnedCredential.getCertificates() != null) {
+          Collection<List<?>> subjectAlternativeNames =
+              returnedCredential.getCertificates()[0].getSubjectAlternativeNames();
+          if (CollectionUtils.isNotEmpty(subjectAlternativeNames)) {
```

Comment:
Any chance of `subjectAlternativeNames` being null?  If so, we should use `isNotBlank` instead.

Reply:
I thought isNotEmpty checked !=null and !.empty() on a collection already.

Reply:
Whoops, looks like `isNotBlank` also accounts for a whitespace only string.  Do we need to worry about that?